cpp/src/arrow/c/bridge_test.cc

```diff
@@ -443,6 +444,13 @@ TEST_F(TestSchemaExport, Union) {
              {ARROW_FLAG_NULLABLE});
 }
 
+TEST_F(TestSchemaExport, RunEndEncoded) {
+  TestNested(run_end_encoded(int16(), uint8()), {"+r", "s", "C"},
+             {"", "run_ends", "values"}, {ARROW_FLAG_NULLABLE, 0, ARROW_FLAG_NULLABLE});
+  TestNested(run_end_encoded(int64(), int32()), {"+r", "l", "i"},
```

Comment:
Just to prove it's not a problem, I'd like to see a test or two where the values are non-primitive (IE nested and/or dictionary encoded)

Reply:
The best I can do is a string array because `run_end_encode` doesn't produce REE-encoded arrays of nested values.

https://github.com/apache/arrow/blob/main/cpp/src/arrow/compute/kernels/vector_run_end_encode.cc#L601-L621

My reasoning for de-prioritizing that was that gains from run-end encoding complex objects (lists, structs) are less likely.

Reply:
But if you mean just for schema export (not the whole array round-trip), I can add any type. :D

Reply:
Of course. The default factories in these tests made me forget about that. Writing tests with REEs of lists of strings now.

Reply:
I added checks on the array roundtrip to forbid nested REEs and REEs with non-supported run-end types